proposals/collection-expression-better-conversion.md

```diff
@@ -0,0 +1,125 @@
+# Better conversion from collection expression element
+
+## Summary
+
+Updates to the better conversion rules to be more consistent with `params`, and better handle current ambiguity scenarios. For example, `ReadOnlySpan<string>` vs `ReadOnlySpan<object>` can currently
+cause ambiguities during overload resolution for `[""]`.
+
+## Detailed Design
+
+The following are the better conversion from expression rules. These replace the rules in https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#overload-resolution.
+
+These rules are:
+
+> Given an implicit conversion `C₁` that converts from an expression `E` to a type `T₁`, and an implicit conversion `C₂` that converts from an expression `E` to a type `T₂`, `C₁` is a ***better conversion*** than `C₂` if one of the following holds:
+> **`E` is a *collection expression*, and `C₁` is a ***better collection conversion from expression***, or**
+> * **`E` is not a *collection expression* and one of the following holds:**
+>   * `E` exactly matches `T₁` and `E` does not exactly match `T₂`
+>   * `E` exactly matches both or neither of `T₁` and `T₂`, and `T₁` is a [*better conversion target*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#11646-better-conversion-target) than `T₂`
+> * `E` is a method group, ...
+
+We add a new definition for ***better collection conversion from expression***, as follows:
+
+Given:
+- `E` is a collection expression with element expressions `[EL₁, EL₂, ..., ELₙ]`
+- `T₁` and `T₂` are collection types
+- `T'₁`:
+    - If `T₁` is a generic type, `T₁` with all type parameters unsubstituted
+    - If `T₁` is an array type, an unsubstituted array type
+    - Otherwise, `T₁`
+- `T'₂` is the same as `T'₁`, except for using `T₂` instead of `T₁`
+- `E₁` is the element type of `T₁`
+- `E₂` is the element type of `T₂`
+- `CE₁ᵢ` are the series of conversions from `ELᵢ` to `E₁`
+- `CE₂ᵢ` are the series of conversions from `ELᵢ` to `E₂`
+
+`T'₁` is at least as good as `T'₂` if there is an identity conversion between them.  
+`T'₁` is better than `T'₂` if:
+- `T'₁` is ReadOnlySpan<T>, and `T'₂` is not ReadOnlySpan<T>
+- `T'₂` is Span<T>, and `T'₂` is an *array_or_array_interface*
+- `T'₂` is implicitly convertible to `T'₁`, and `T'₁` is not implicitly convertible to `T'₂`.
```

Comment:
Should `T'₁` and `T'₂` be reversed here? For example, we should prefer `List<T>` over `IEnumerable<T>`, and `List<T>` is implicitly convertible to `IEnumerable<T>`.

```suggestion
- `T'₁` is implicitly convertible to `T'₂`, and `T'₂` is not implicitly convertible to `T'₁`.
```